external/fv3fit/fv3fit/emulation/thermobasis/emulator.py

```diff
@@ -0,0 +1,366 @@
+from collections import defaultdict
+import dataclasses
+import json
+import logging
+from typing import (
+    Optional,
+    Sequence,
+    Union,
+    List,
+)
+import os
+import numpy
+import tensorflow as tf
+import dacite
+from fv3fit.emulation.thermobasis.batch import (
+    get_prognostic_variables,
+    batch_to_specific_humidity_basis,
+)
+from fv3fit.emulation.thermobasis.loggers import (
+    WandBLogger,
+    ConsoleLogger,
+    TBLogger,
+    LoggerList,
+)
+from fv3fit.emulation.thermobasis.loss import RHLoss, QVLoss, MultiVariableLoss
+from fv3fit.emulation.thermobasis.models import (
+    UVTQSimple,
+    UVTRHSimple,
+    ScalarMLP,
+    RHScalarMLP,
+)
+from fv3fit.emulation.thermobasis.models import V1QCModel
+
+
+def average(metrics):
+    return {key: sum(metrics[key]) / len(metrics[key]) for key in metrics}
+
+
+@dataclasses.dataclass
+class BatchDataConfig:
+    training_path: str
+    testing_path: str
+
+
+@dataclasses.dataclass
+class Config:
+    """
+    Attrs:
+        batch_size: the largest batch size to be used for a single gradient
+            descent step
+        learning_rate: the learning rate for each gradient descent step
+        online: whether predictions are used or not
+        train: if True the model is trained online
+        batch: if provided then these data are used for training the ML model
+        num_hidden_layers: number of hidden layers used. Only implemented for
+            QVLoss targets.
+        wandb_logger: if True, then enable weights and biases saving
+
+    """
+
+    batch_size: int = 64
+    learning_rate: float = 0.01
+    num_hidden: int = 256
+    num_hidden_layers: int = 1
+    momentum: float = 0.5
+
+    # other parameters
+    extra_input_variables: List[str] = dataclasses.field(default_factory=list)
+    epochs: int = 1
+    levels: int = 79
+    batch: Optional[BatchDataConfig] = None
+    target: Union[MultiVariableLoss, QVLoss, RHLoss] = dataclasses.field(
+        default_factory=MultiVariableLoss
+    )
+    relative_humidity: bool = False
+    wandb_logger: bool = False
+
+    output_path: str = ""
+    weight_sharing: bool = False
+    cloud_water: bool = False
+
+    @property
+    def input_variables(self) -> List[str]:
```

Comment:
Unrelated, but would you like to do the PR refactoring the regular fv3fit training to use Hyperparameters classes for its variable listings, or should I pick that up?

Reply:
Feel free to pick it up.